File: public/templates/library/data/merge_mdna_course.py
#!/usr/bin/env python3
"""One-off: Merge mDNA course (with weeks) into alan-hirsch.json contentLibrary.courses."""
import json
from pathlib import Path

data_dir = Path(__file__).resolve().parent
alan_path = data_dir / "alan-hirsch.json"
mdna_path = data_dir / "courses" / "mdna.json"

alan = json.loads(alan_path.read_text(encoding="utf-8"))
mdna = json.loads(mdna_path.read_text(encoding="utf-8"))

# Preserve lesson titles from existing mdna in alan-hirsch (e.g. Alan-voice titles)
existing_mdna = next((c for c in alan["contentLibrary"]["courses"] if c.get("slug") == "mdna"), None)
if existing_mdna and existing_mdna.get("weeks"):
    for wi, week in enumerate(mdna.get("weeks", [])):
        exist_week = next((w for w in existing_mdna["weeks"] if w.get("index") == week.get("index")), None)
        if not exist_week:
            continue
        for li, lesson in enumerate(week.get("lessons", [])):
            exist_lessons = exist_week.get("lessons") or []
            if li < len(exist_lessons) and exist_lessons[li].get("title"):
                lesson["title"] = exist_lessons[li]["title"]

existing = [c for c in alan["contentLibrary"]["courses"] if c.get("slug") != "mdna"]
alan["contentLibrary"]["courses"] = [mdna] + existing

alan_path.write_text(json.dumps(alan, indent=2, ensure_ascii=False) + "\n", encoding="utf-8")
print("Merged mDNA course into alan-hirsch.json (courses[0])")
print("Course has", len(mdna.get("weeks", [])), "weeks")

File: public/templates/library/data/merge_mdna_course.js
/**
 * One-off: Merge mDNA course (with weeks) into alan-hirsch.json contentLibrary.courses.
 * Run from directories/data: node merge_mdna_course.js
 */
const fs = require("fs");
const path = require("path");

const dataDir = __dirname;
const alanPath = path.join(dataDir, "alan-hirsch.json");
const mdnaPath = path.join(dataDir, "courses", "mdna.json");

const alan = JSON.parse(fs.readFileSync(alanPath, "utf8"));
const mdna = JSON.parse(fs.readFileSync(mdnaPath, "utf8"));

const existing = alan.contentLibrary.courses.filter((c) => c.slug !== "mdna");
alan.contentLibrary.courses = [mdna, ...existing];

fs.writeFileSync(alanPath, JSON.stringify(alan, null, 2) + "\n", "utf8");
console.log("Merged mDNA course into alan-hirsch.json (courses[0])");
console.log("Course has", mdna.weeks?.length ?? 0, "weeks");
